huaweicloud/services/iam/data_source_huaweicloud_identity_permissions.go

```diff
@@ -0,0 +1,156 @@
+package iam
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk/openstack/identity/v3/roles"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
+)
+
+func DataSourceIdentityPermissions() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIdentityPermissionsRead,
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "system",
+				ValidateFunc: validation.StringInSlice([]string{"system", "system-role", "system-policy", "custom"}, false),
+			},
+			"catalog": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"permissions": {
+				Type:     schema.TypeList,
+				Elem:     iamPermissionSchema(),
+				Computed: true,
+			},
+		},
+	}
+}
+
+func iamPermissionSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description_cn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"catalog": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+	return &sc
+}
+
+// dataSourceIdentityPermissionsRead performs the permissions lookup
+func dataSourceIdentityPermissionsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	identityClient, err := cfg.IdentityV3Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating identity client: %s", err)
+	}
+
+	listOpts := roles.ListOpts{
+		DisplayName: d.Get("name").(string),
+		Catalog:     d.Get("catalog").(string),
+	}
+
+	roleType := d.Get("type").(string)
+	if roleType == "custom" {
+		if cfg.DomainID == "" {
+			return diag.Errorf("the domain_id must be specified in the provider configuration")
```

Comment:
`domain_id` is not displayed on our docs, better to use `domain_name` here.

Reply:
It is only happens in the internal environment.
The `domain_id` parameter will make the provider don't call the API which is not exist in the internal environment.